recipes/_omnibus_toolchain.rb

```diff
@@ -17,12 +17,44 @@
 # limitations under the License.
 #
 
-log "omnibus-toolchain is #{omnibus_toolchain_enabled? ? 'enabled' : 'disabled'}"
-return unless omnibus_toolchain_enabled?
+if windows?
+  toolchain_options = {
+    product_name: node['omnibus']['toolchain_name'],
+    product_version: node['omnibus']['toolchain_version'],
+    channel: node['omnibus']['toolchain_channel'].to_sym,
+    # Override detected architecture on Windows as we use properly
+    # configured 64-bit instances to compile things in 32-bit mode.
+    architecture: (windows_arch_i386? ? 'i386' : 'x86_64'),
+  }
+
+  artifact_info = mixlib_install_artifact_info_for(toolchain_options)
+
+  remote_artifact_path = artifact_info.url
+  local_artifact_path  = ::File.join(Chef::Config[:file_cache_path], ::File.basename(remote_artifact_path))
+
+  remote_file local_artifact_path do
+    source remote_artifact_path
+    mode '0644'
+    checksum artifact_info.sha256
+    backup 1
+  end
+
+  omnibus_env['MSYSTEM'] << mingw_toolchain_name.upcase
+  omnibus_env['OMNIBUS_WINDOWS_ARCH'] << (windows_arch_i386? ? 'x86' : 'x64')
+  omnibus_env['BASH_ENV'] << msys2_path(windows_safe_path_join(toolchain_install_dir, 'embedded', 'bin', 'etc', 'msys2.bashrc'))
```

Comment:
Just to close the loop on a Zoom conversation @tduffield and I had. We should remove the `msys2_path` helper from this completely as it should not be needed and fixing it to properly convert all style windows paths (e.g. `C:\foo`, `C:/foo`) to a unix style path (`/c/foo`) could actually break things as `BASH_ENV` is being set to `C:\opscode\omnibus-toolchain\embedded\bin\etc\msys2.bashrc` currently.